GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
```

Comment:
Precondition usually used for checking some condition before allowing to go into its main action. If the condition is not met, the whole log of task will abort. However, our new class introduced here seems just a collection of utility methods.
I got pointer from Chao about this Guava’s Preconditions, and taking a look, it does some state checking, I guess this is its major role, as well as general argument checking methods. Anyway, it is Chao’s decision for the class name, but I think the big name can invite unnecessary confusions.